### PR TITLE
setup UserMailer to send welcome email, testing links in emails still…

### PR DESCRIPTION
**Pull request by MatthewReuther**
Implemented UserMailer; letter_opener is not executing  on all computers.

### CODE REVIEW

**Comment by MatthewReuther** [comment]
Postmark api key needed, please.

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class ApplicationMailer < ActionMailer::Base
+  default from: "welcome@median.com"
+  layout 'mailer'
+end
+
```
Comment (elizamarcum) [inline app/mailers/application_mailer.rb]: This whitespace seems accidental

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class UserMailer < ApplicationMailer
+
+  def welcome_email(user)
+    @user = user
+    @url = 'http://median.com/user'
```
Comment (elizamarcum) [inline app/mailers/user_mailer.rb]: This URL shouldn't be hardcoded.  Also, I don't think it is used.

File: app/views/layouts/mailer.html.haml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+%hmtl
+  %body
```
Comment (elizamarcum) [inline app/views/layouts/mailer.html.haml]: I'm not sure this is used.

File: app/views/layouts/mailer.text.haml [hunk 4]
```diff
@@ -0,0 +1 @@
+= yield
```
Comment (elizamarcum) [inline app/views/layouts/mailer.text.haml]: I don't think this is necessary.

File: config/database.yml.example [hunk 5]
```diff
@@ -1,85 +0,0 @@
-# PostgreSQL. Versions 8.2 and up are supported.
```
Comment (elizamarcum) [inline config/database.yml.example:1]: This file shouldn't be deleted.

File: spec/mailers/previews/user_mailer_preview.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Preview all emails at http://localhost:3000/rails/mailers/user_mailer
+class UserMailerPreview < ActionMailer::Preview
```
Comment (elizamarcum) [inline spec/mailers/previews/user_mailer_preview.rb]: This file isn't used.

File: spec/mailers/user_mailer_spec.rb [hunk 7]
```diff
@@ -0,0 +1,5 @@
+require "rails_helper"
+
```
Comment (elizamarcum) [inline spec/mailers/user_mailer_spec.rb]: This file isn't used.

**Comment by elizamarcum** [comment]
@MatthewReuther I'll add the Postmark key when I merge in the changes from one of the branches that has figaro configured.

**Comment by elizamarcum** [comment]
@MatthewReuther / @mollytaryn This is looking pretty good but there are still some previous comments to address.  You can see them by looking in the "Files Changed" tab.

**Comment by MatthewReuther** [comment]
Went back and fixed issues from previous comments.

File: config/environments/development.rb [hunk 8]
```diff
@@ -12,10 +12,14 @@
   # Show full error reports and disable caching.
   config.consider_all_requests_local       = true
   config.action_controller.perform_caching = false
+  config.action_mailer.delivery_method = :letter_opener
 
   # Don't care if the mailer can't send.
   config.action_mailer.raise_delivery_errors = false
 
+  # Letter Opener action method
+  config.action_mailer.delivery_method = :letter_opener
```
Comment (elizamarcum) [inline config/environments/development.rb:20]: You've added letter_opener twice.

**Comment by MatthewReuther** [comment]
Removed extra letter_opener.

**Pull request merged**